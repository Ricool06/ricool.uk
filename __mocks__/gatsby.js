import React from 'react';

const gatsby = jest.requireActual('gatsby');

module.exports = {
  ...gatsby,
  graphql: jest.fn(queryString => queryString),
  Link: jest.fn().mockImplementation(({ to, ...rest }) => React.createElement('a', {
    ...rest,
    href: to,
  })),
  StaticQuery: jest.fn(),
};
