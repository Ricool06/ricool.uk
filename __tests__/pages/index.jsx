import React from 'react';

import { StaticQuery } from 'gatsby';
import renderer from 'react-test-renderer';
import IndexPage from '../../src/pages/index';

describe('IndexPage', () => {
  const staticQueryData = {
    site: {
      siteMetadata: {
        title: 'Ricool\'s Blog',
        description: 'making things all day',
        author: 'Ricool06',
      },
    },
  };

  it('should have a correctly configured StaticQuery component for default export', () => {
    StaticQuery.mockImplementation(({ render: mockRender }) => mockRender(staticQueryData));

    const data = {
      allMarkdownRemark: {
        edges: [
          {
            node: {
              id: 'c34acf72-ec77-5252-8e8e-61d030108cbb',
              excerpt: 'The start of some blog entry...',
              frontmatter: {
                date: 'February 10, 2019',
                path: '/work-based-learning-week-1',
                title: 'My first blog post',
              },
            },
          },
          {
            node: {
              id: 'c34acf72-ec77-5252-8e8e-61d0aaaaaaaa',
              excerpt: 'The start of some other blog entry...',
              frontmatter: {
                date: 'February 11, 2019',
                path: '/work-based-learning-week-2',
                title: 'My second blog post',
              },
            },
          },
        ],
      },
    };
    const jsonSnapshot = renderer.create(<IndexPage data={data} />).toJSON();

    expect(jsonSnapshot).toMatchSnapshot();
  });
});
