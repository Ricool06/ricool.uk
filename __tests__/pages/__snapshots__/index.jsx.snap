// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexPage should have a correctly configured StaticQuery component for default export 1`] = `
Array [
  <div
    style={
      Object {
        "background": "#339988",
        "marginBottom": "1.45rem",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "margin": "0 auto",
          "maxWidth": 960,
          "padding": "1.45rem 1.0875rem",
        }
      }
    >
      <h1
        style={
          Object {
            "flex": "1 1 auto",
            "margin": 0,
          }
        }
      >
        <a
          href="/"
          style={
            Object {
              "color": "white",
              "textDecoration": "none",
            }
          }
        >
          Ricool's Blog
        </a>
      </h1>
      <div />
    </div>
  </div>,
  <div
    style={
      Object {
        "margin": "0 auto",
        "maxWidth": 960,
        "padding": "0px 1.0875rem 1.45rem",
        "paddingTop": 0,
      }
    }
  >
    <div>
      <div
        className="sc-bdVaJa crvamO"
      >
        <a
          href="/work-based-learning-week-1"
        >
          <h2>
            My first blog post
          </h2>
          <div
            className="post-content"
          >
            <h4>
              February 10, 2019
            </h4>
            <p>
              The start of some blog entry...
            </p>
          </div>
          <hr />
        </a>
      </div>
      <div
        className="sc-bdVaJa crvamO"
      >
        <a
          href="/work-based-learning-week-2"
        >
          <h2>
            My second blog post
          </h2>
          <div
            className="post-content"
          >
            <h4>
              February 11, 2019
            </h4>
            <p>
              The start of some other blog entry...
            </p>
          </div>
          <hr />
        </a>
      </div>
    </div>
    <footer>
      Â©
      2019
      , Built with
       
      <a
        href="https://www.gatsbyjs.org"
      >
        Gatsby
      </a>
    </footer>
  </div>,
]
`;
